AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -174,26 +174,30 @@ public final class APIClient: APIClientProtocol {
         _ request: R
     ) throws -> HTTPResponse<R.ResponseType> {
         log(result: result, request: request)
+        
         do {
             if result.data.isEmpty, let emptyResponse = EmptyResponse() as? R.ResponseType {
                 return HTTPResponse(
                     headers: result.headers,
                     statusCode: result.statusCode,
-                    responseBody: emptyResponse
+                    responseBody: emptyResponse,
+                    responseData: result.data
                 )
             } else {
                 return HTTPResponse(
                     headers: result.headers,
                     statusCode: result.statusCode,
-                    responseBody: try coder.decode(R.ResponseType.self, from: result.data)
+                    responseBody: try coder.decode(R.ResponseType.self, from: result.data),
+                    responseData: result.data
                 )
             }
         } catch {
             if let errorResponse = try? coder.decode(R.ErrorResponseType.self, from: result.data) {
                 throw HTTPErrorResponse(
                     headers: result.headers,
                     statusCode: result.statusCode,
-                    responseBody: errorResponse
+                    responseBody: errorResponse,
+                    responseData: result.data
                 )
             } else if let decodingError = error as? DecodingError {
                 throw ParsingError(
```

Comment:
Passing `result.data` to 'ParsingError' could be a good idea as well, so consumer can try to figure out what is going wrong 🤔

Reply:
Yeah that's a nice idea indeed, will update to have ParsingError conform to `AnyDataResponse` and set the responseData on it